celer/dropin_sklearn.py

```diff
@@ -276,13 +279,269 @@ def _more_tags(self):
         return {'multioutput': False}
 
 
+class AdaptiveLasso(Lasso):
```

Comment:
I am now thinking this is a bad name, as adaptive Lasso takes weights equal to 1 / w^ols_j, and it performs a single Lasso.
What we implement is rather iterative l1 reweighting (Candes Wakin Boyd paper).

IterativeReweightedLasso seems more correct for me, but it does not pop up on google, and I don't know if it good for visibility. When people talk about it in scikit-learn, they say adaptive lasso : https://github.com/scikit-learn/scikit-learn/pull/4912
@agramfort @josephsalmon do you have an opinion ?

Reply:
The main naming issue to me is that an estimator should be well-defined mathematically; then the implementation is something under the hood for the user.

Here, the algorithm you are proposing is a DC-programming (à la Gasso et al. 2009) approach for solving sparse regression with \ell_0.5 regularization (also referred to as reweighted l1 by Cands et al.).
Hence, I would be in favor of separating the "theoretical" estimator from the algorithms used (for instance a coordinate descent alternative could be considered as another solver for sparse regression with \ell_0.5 regularization).

I agree that AdaptiveLasso is originally 2-step:
1) OLS
2) Lasso reweighted with weights controlled by step 1.

But I think this is vague enough in the original article (any consistent estimator can be used in the first step, not only OLS), so we can reuse this naming.

Potentially, the exponent \gamma (corresponding to the \ell_q norm used in the Adaptive Lasso paper) could be an optional parameter something like:

`lq_norm = 0.5`

(with the possibility to add more variants later on).
 
So in the end, I won't bother too much about the naming and stick to AdaptiveLasso as a good shortcut.